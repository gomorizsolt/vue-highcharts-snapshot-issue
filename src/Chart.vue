<template>
  <chart :options="chartOptions"></chart>
</template>

<script>
import { Chart } from "highcharts-vue";

export default {
  components: {
    chart: Chart,
  },
  data() {
    return {
      chartOptions: {
        chart: {
          type: "column",
        },
        series: [
          {
            name: "iPhone 11 Pro",
            data: [{ name: "iPhone 11 Pro", y: 18, percentage: "18%" }],
          },
          {
            name: "iPhone 8",
            data: [{ name: "iPhone 8", y: 14, percentage: "14%" }],
          },
          {
            name: "iPhone backup",
            data: [{ name: "iPhone backup", y: 16, percentage: "16%" }],
          },
          {
            name: "scale up app iPhone",
            data: [{ name: "scale up app iPhone", y: 7, percentage: "7%" }],
          },
          {
            name: "iPhone reset",
            data: [{ name: "iPhone reset", y: 9, percentage: "9%" }],
          },
        ],
      },
    };
  },
};
</script>
