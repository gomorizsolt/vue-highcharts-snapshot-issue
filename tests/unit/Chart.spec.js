import { mount } from "@vue/test-utils";
import Highcharts from "highcharts";
import Chart from "@/Chart.vue";

const data = [
  {
    name: "Distribution in %",
    data: [
      { name: "iPhone 11 Pro", y: 18, percentage: "18%" },
      { name: "iPhone 11 Pro", y: 18, percentage: "18%" },
      { name: "iPhone 11 Pro", y: 18, percentage: "18%" },
      { name: "iPhone 8", y: 14, percentage: "14%" },
      { name: "iPhone backup", y: 16, percentage: "16%" },
      { name: "scale up app iPhone", y: 7, percentage: "7%" },
      { name: "iPhone reset", y: 9, percentage: "9%" },
    ],
  },
];

Highcharts.useSerialIds(true);

describe("Chart.vue", () => {
  it("renders correctly", () => {
    const wrapper = mount(Chart, {
      propsData: {
        type: "bar",
        data,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
