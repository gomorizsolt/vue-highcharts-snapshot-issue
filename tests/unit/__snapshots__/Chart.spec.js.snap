// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart.vue renders correctly 1`] = `
<div type="bar" data="[object Object]" data-highcharts-chart="0" style="overflow: hidden;">
  <div id="highcharts-0" style="position: relative; overflow: hidden; width: 600px; height: 400px; text-align: left; line-height: normal; z-index: 0; user-select: none;" dir="ltr" class="highcharts-container "><svg version="1.1" class="highcharts-root" style="font-family:&quot;Lucida Grande&quot;, &quot;Lucida Sans Unicode&quot;, Arial, Helvetica, sans-serif;font-size:12px;" xmlns="http://www.w3.org/2000/svg" width="600" height="400" viewBox="0 0 600 400">
      <desc>Created with Highcharts 8.2.0</desc>
      <defs>
        <clipPath id="highcharts-1-">
          <rect x="0" y="0" width="NaN" height="NaN" fill="none"></rect>
        </clipPath>
      </defs>
      <rect fill="#ffffff" class="highcharts-background" x="0" y="0" width="600" height="400" rx="0" ry="0"></rect>
      <rect fill="none" class="highcharts-plot-background" x="NaN" y="10" width="NaN" height="NaN"></rect>
      <g class="highcharts-grid highcharts-xaxis-grid" data-z-index="1">
        <path fill="none" data-z-index="1" class="highcharts-grid-line" d="M 0 0" opacity="1"></path>
      </g>
      <g class="highcharts-grid highcharts-yaxis-grid" data-z-index="1">
        <path fill="none" stroke="#e6e6e6" stroke-width="1" data-z-index="1" class="highcharts-grid-line" d="M 0 0" opacity="1"></path>
        <path fill="none" stroke="#e6e6e6" stroke-width="1" data-z-index="1" class="highcharts-grid-line" d="M 0 0" opacity="1"></path>
        <path fill="none" stroke="#e6e6e6" stroke-width="1" data-z-index="1" class="highcharts-grid-line" d="M 0 0" opacity="1"></path>
        <path fill="none" stroke="#e6e6e6" stroke-width="1" data-z-index="1" class="highcharts-grid-line" d="M 0 0" opacity="1"></path>
        <path fill="none" stroke="#e6e6e6" stroke-width="1" data-z-index="1" class="highcharts-grid-line" d="M 0 0" opacity="1"></path>
      </g>
      <rect fill="none" class="highcharts-plot-border" data-z-index="1" x="0" y="10" width="0" height="0"></rect>
      <g class="highcharts-axis highcharts-xaxis" data-z-index="2">
        <path fill="none" class="highcharts-tick" stroke="#ccd6eb" stroke-width="1" d="M 0 0" opacity="1"></path>
        <path fill="none" class="highcharts-axis-line" stroke="#ccd6eb" stroke-width="1" data-z-index="7" d="M 0 0"></path>
      </g>
      <g class="highcharts-axis highcharts-yaxis" data-z-index="2"><text x="0" data-z-index="7" text-anchor="middle" transform="translate(0,0) rotate(270 0 -9999)" class="highcharts-axis-title" style="color:#666666;fill:#666666;" y="-9999">
          <tspan>Values</tspan>
        </text>
        <path fill="none" class="highcharts-axis-line" data-z-index="7" d="M 0 0"></path>
      </g>
      <g class="highcharts-series-group" data-z-index="3">
        <g class="highcharts-series highcharts-series-0 highcharts-column-series highcharts-color-0 highcharts-tracker" data-z-index="0.1" opacity="1" transform="translate(0,10) scale(1 0.001)" clip-path="url(#highcharts-1-)">
          <rect x="NaN" y="NaN" width="NaN" height="NaN" fill="#7cb5ec" stroke="#ffffff" stroke-width="1" opacity="1" class="highcharts-point highcharts-color-0"></rect>
        </g>
        <g class="highcharts-markers highcharts-series-0 highcharts-column-series highcharts-color-0" data-z-index="0.1" opacity="1" transform="translate(0,10) scale(1 1)"></g>
        <g class="highcharts-series highcharts-series-1 highcharts-column-series highcharts-color-1 highcharts-tracker" data-z-index="0.1" opacity="1" transform="translate(0,10) scale(1 0.001)" clip-path="url(#highcharts-1-)">
          <rect x="NaN" y="NaN" width="NaN" height="NaN" fill="#434348" stroke="#ffffff" stroke-width="1" opacity="1" class="highcharts-point highcharts-color-1"></rect>
        </g>
        <g class="highcharts-markers highcharts-series-1 highcharts-column-series highcharts-color-1" data-z-index="0.1" opacity="1" transform="translate(0,10) scale(1 1)"></g>
        <g class="highcharts-series highcharts-series-2 highcharts-column-series highcharts-color-2 highcharts-tracker" data-z-index="0.1" opacity="1" transform="translate(0,10) scale(1 0.001)" clip-path="url(#highcharts-1-)">
          <rect x="NaN" y="NaN" width="NaN" height="NaN" fill="#90ed7d" stroke="#ffffff" stroke-width="1" opacity="1" class="highcharts-point highcharts-color-2"></rect>
        </g>
        <g class="highcharts-markers highcharts-series-2 highcharts-column-series highcharts-color-2" data-z-index="0.1" opacity="1" transform="translate(0,10) scale(1 1)"></g>
        <g class="highcharts-series highcharts-series-3 highcharts-column-series highcharts-color-3 highcharts-tracker" data-z-index="0.1" opacity="1" transform="translate(0,10) scale(1 0.001)" clip-path="url(#highcharts-1-)">
          <rect x="NaN" y="NaN" width="NaN" height="NaN" fill="#f7a35c" stroke="#ffffff" stroke-width="1" opacity="1" class="highcharts-point highcharts-color-3"></rect>
        </g>
        <g class="highcharts-markers highcharts-series-3 highcharts-column-series highcharts-color-3" data-z-index="0.1" opacity="1" transform="translate(0,10) scale(1 1)"></g>
        <g class="highcharts-series highcharts-series-4 highcharts-column-series highcharts-color-4 highcharts-tracker" data-z-index="0.1" opacity="1" transform="translate(0,10) scale(1 0.001)" clip-path="url(#highcharts-1-)">
          <rect x="NaN" y="NaN" width="NaN" height="NaN" fill="#8085e9" stroke="#ffffff" stroke-width="1" opacity="1" class="highcharts-point highcharts-color-4"></rect>
        </g>
        <g class="highcharts-markers highcharts-series-4 highcharts-column-series highcharts-color-4" data-z-index="0.1" opacity="1" transform="translate(0,10) scale(1 1)"></g>
      </g><text x="300" text-anchor="middle" class="highcharts-title" data-z-index="4" style="color:#333333;font-size:18px;fill:#333333;" y="24">
        <tspan>Chart title</tspan>
      </text><text x="300" text-anchor="middle" class="highcharts-subtitle" data-z-index="4" style="color:#666666;fill:#666666;" y="10"></text><text x="10" text-anchor="start" class="highcharts-caption" data-z-index="4" style="color:#666666;fill:#666666;" y="397"></text>
      <g class="highcharts-legend" data-z-index="7" transform="translate(300,362)">
        <rect fill="none" class="highcharts-legend-box" rx="0" ry="0" visibility="visible"></rect>
        <g data-z-index="1">
          <g>
            <g class="highcharts-legend-item highcharts-column-series highcharts-color-0 highcharts-series-0" data-z-index="1" transform="translate(8,3)"><text x="21" style="color:#333333;cursor:pointer;font-size:12px;font-weight:bold;fill:#333333;" text-anchor="start" data-z-index="2" y="15">
                <tspan>iPhone 11 Pro</tspan>
              </text>
              <rect x="2" y="4" width="12" height="12" fill="#7cb5ec" rx="6" ry="6" class="highcharts-point" data-z-index="3"></rect>
            </g>
            <g class="highcharts-legend-item highcharts-column-series highcharts-color-1 highcharts-series-1" data-z-index="1" transform="translate(0,3)"><text x="21" y="15" style="color:#333333;cursor:pointer;font-size:12px;font-weight:bold;fill:#333333;" text-anchor="start" data-z-index="2">
                <tspan>iPhone 8</tspan>
              </text>
              <rect x="2" y="4" width="12" height="12" fill="#434348" rx="6" ry="6" class="highcharts-point" data-z-index="3"></rect>
            </g>
            <g class="highcharts-legend-item highcharts-column-series highcharts-color-2 highcharts-series-2" data-z-index="1" transform="translate(0,3)"><text x="21" y="15" style="color:#333333;cursor:pointer;font-size:12px;font-weight:bold;fill:#333333;" text-anchor="start" data-z-index="2">
                <tspan>iPhone backup</tspan>
              </text>
              <rect x="2" y="4" width="12" height="12" fill="#90ed7d" rx="6" ry="6" class="highcharts-point" data-z-index="3"></rect>
            </g>
            <g class="highcharts-legend-item highcharts-column-series highcharts-color-3 highcharts-series-3" data-z-index="1" transform="translate(0,3)"><text x="21" y="15" style="color:#333333;cursor:pointer;font-size:12px;font-weight:bold;fill:#333333;" text-anchor="start" data-z-index="2">
                <tspan>scale up app iPhone</tspan>
              </text>
              <rect x="2" y="4" width="12" height="12" fill="#f7a35c" rx="6" ry="6" class="highcharts-point" data-z-index="3"></rect>
            </g>
            <g class="highcharts-legend-item highcharts-column-series highcharts-color-4 highcharts-series-4" data-z-index="1" transform="translate(0,3)"><text x="21" y="15" style="color:#333333;cursor:pointer;font-size:12px;font-weight:bold;fill:#333333;" text-anchor="start" data-z-index="2">
                <tspan>iPhone reset</tspan>
              </text>
              <rect x="2" y="4" width="12" height="12" fill="#8085e9" rx="6" ry="6" class="highcharts-point" data-z-index="3"></rect>
            </g>
          </g>
        </g>
      </g>
      <g class="highcharts-axis-labels highcharts-xaxis-labels" data-z-index="7"><text x="0" style="color:#666666;cursor:default;font-size:11px;fill:#666666;" text-anchor="middle" transform="translate(0,0)">0</text></g>
      <g class="highcharts-axis-labels highcharts-yaxis-labels" data-z-index="7"><text x="0" style="color:#666666;cursor:default;font-size:11px;fill:#666666;" text-anchor="end" transform="translate(0,0)">0</text><text x="0" style="color:#666666;cursor:default;font-size:11px;fill:#666666;" text-anchor="end" transform="translate(0,0)">5</text><text x="0" style="color:#666666;cursor:default;font-size:11px;fill:#666666;" text-anchor="end" transform="translate(0,0)">10</text><text x="0" style="color:#666666;cursor:default;font-size:11px;fill:#666666;" text-anchor="end" transform="translate(0,0)">15</text><text x="0" style="color:#666666;cursor:default;font-size:11px;fill:#666666;" text-anchor="end" transform="translate(0,0)">20</text></g><text x="590" class="highcharts-credits" text-anchor="end" data-z-index="8" style="cursor:pointer;color:#999999;font-size:9px;fill:#999999;" y="395">Highcharts.com</text>
    </svg></div>
</div>
`;
